### PR TITLE
Identify network interfaces by mac over name

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Currently we rely on network interface name to identify devices and tie them to config. This is problematic because they can change. And regularly do on some supported systems.

This PR ties the config to the mac address of the device instead of the name as that is far less likely to change. We believe it is unchanging for the boards we support currently although an operating system fix is possible if we do add support for a board where it can change.

This PR is backwards compatible API-wise. It adds support for updating a network interface identified by mac address but still allows clients to identify the one to update by name as well.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/operating-system/issues/2174
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by pvizeli — APPROVED** [review]

**Comment by pvizeli** [comment]
I think we should prefer inet name on our API, but it's nice additional that you also can address the interface over mac

File: supervisor/host/configuration.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
+"""Network objects for host manager."""
```
Comment (agners) [inline supervisor/host/configuration.py:1]: These are still very much network related configurations, maybe `host.network.configuration`?

**Review by agners — COMMENTED** [review]
How/when will the migration take place? From w hat I understand whenever the configuration gets saved the migration will take place, but will we force a config save somehow?

**Comment by pvizeli** [comment]
> How/when will the migration take place? From w hat I understand whenever the configuration gets saved the migration will take place, but will we force a config save somehow?

On startup of this supervisor, it get migrated automatic

**Comment by agners** [comment]
> On startup of this supervisor, it get migrated automatic

From what I remember last time we worked on the networking, we disabled the automatic save "unless necessary" for some reason. I don't remember the exact details, was just wondering if we have to do something there to trigger the automatic save.

**Pull request merged**